packages/forms/src/UIForm/UIForm.component.js

```diff
@@ -46,11 +44,31 @@ export default class UIForm extends React.Component {
 	onChange(event, { schema, value }) {
 		const payload = {
 			formName: this.props.formName,
-			properties: this.props.properties,
+			properties: this.props.properties || {},
 			schema,
 			value,
 		};
-		this.props.onChange(event, payload);
+		payload.formData = mutateValue(payload.properties, schema.key, value);
+		if (this.props.moz) {
+			this.props.onChange(payload);
+		} else {
+			this.props.onChange(event, payload);
+		}
+		// trigger if value is correct
+		if (schema.triggers && schema.triggers.length) {
+			let formData = payload.formData;
+			let propertyName = schema.key.join('.');
+			if (this.props.moz) {
+				schema.key.forEach((key, index) => {
+					if (index !== schema.key.length - 1) {
+						formData = formData[key];
```

Comment:
What is that ? on old forms, the trigger get the parent object value instead of the value ?

Reply:
yes :(
Sorry for that it's for compatibility and change that need to update all forms.
Worse we have also backend change to support this use case.

Reply:
wow this is a weird choice